app/src/main/java/org/oppia/android/app/customview/PromotedStoryCardView.kt

```diff
@@ -0,0 +1,60 @@
+package org.oppia.android.app.customview
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import com.google.android.material.card.MaterialCardView
+import org.oppia.android.R
+import org.oppia.android.app.model.Spotlight
+import org.oppia.android.app.spotlight.SpotlightManager
+import org.oppia.android.app.spotlight.SpotlightTarget
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.view.ViewComponentFactory
+import org.oppia.android.app.view.ViewComponentImpl
+import javax.inject.Inject
+
+/** Custom view to hold the promoted story cards. */
+class PromotedStoryCardView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : MaterialCardView(context, attrs, defStyleAttr) {
+
+  @Inject
+  lateinit var fragment: Fragment
+
+  @Inject
+  lateinit var resourceHandler: AppLanguageResourceHandler
+
+  private var isSpotlit = false
+
+  /** Sets the index at which this custom view is located inside the recycler view. */
+  fun setIndex(index: Int) {
+    if (!isSpotlit) {
+      isSpotlit = true
+      val spotlightTarget = SpotlightTarget(
+        this,
+        resourceHandler.getStringInLocale(R.string.promoted_story_spotlight_hint),
+        feature = Spotlight.FeatureCase.PROMOTED_STORIES
+      )
+      if (index == 0) {
```

Comment:
Ditto here--a boolean may be a better fit (and it could be combined with the ``if (!isSpotlit)`` above for slightly cleaner control flow).

Reply:
combining these two conditions together